btk/measure.py

```diff
@@ -218,18 +218,30 @@ def __next__(self):
 
         Returns:
             draw_blend_generator output from `__next__` method.
-            measurement output: List of length `batch_size`, where each element is a list of
+            measurement_results: List of length `batch_size`, where each element is a list of
                                 `len(measure_functions)` corresponding to the measurements made by
                                 each function on each element of the batch.
         """
         blend_output = next(self.draw_blend_generator)
         input_args = ((blend_output, i) for i in range(self.batch_size))
-        measure_results = multiprocess(
+        measure_output = multiprocess(
             self.run_batch,
             input_args,
             self.cpus,
             self.verbose,
         )
         if self.verbose:
             print("Measurement performed on batch")
+        measure_results = {}
+        for i, f in enumerate(self.measure_functions):
+            measure_dic = {}
+            for key in ["catalog", "deblended_images", "segmentation"]:
+                if measure_output[0][i][key] is not None:
+                    measure_dic[key] = [
+                        measure_output[j][i][key] for j in range(len(measure_output))
+                    ]
+            measure_results[f.__name__] = measure_dic
+        if len(self.measure_functions) == 1:
+            measure_results = measure_results[self.measure_functions[0].__name__]
```

Comment:
I see that you have two separate cases depending on how many measurement functions you are using (1 vs >1). This makes is so that we have to split the logic in the metrics code below (and possibly later in the unit tests). I would suggest getting rid of this additional 'if' statement for now. 

Later we can discuss how to make accessing the information less cumbersome for the user/us, but I think splitting the logic now just makes it confusing at least for me. (had to spend sometime figuring out `if` statement in `__next__` from metrics_generator)

Reply:
I don't know, it doesn't feel too cumbersome to me ; but why not.

Reply:
I will commit this change soon